hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoUpdateTransitionLogic.java

```diff
@@ -70,120 +63,61 @@
 public class CryptoUpdateTransitionLogic implements TransitionLogic {
 	private static final Logger log = LogManager.getLogger(CryptoUpdateTransitionLogic.class);
 
-	private static final EnumSet<AccountProperty> EXPIRY_ONLY = EnumSet.of(AccountProperty.EXPIRY);
-
 	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
 
-	private final HederaLedger ledger;
+	private final AccountStore accountStore;
 	private final OptionValidator validator;
 	private final TransactionContext txnCtx;
+	private final TransactionRecordService transactionRecordService;
 	private final GlobalDynamicProperties dynamicProperties;
 
 	@Inject
 	public CryptoUpdateTransitionLogic(
-			HederaLedger ledger,
+			AccountStore accountStore,
 			OptionValidator validator,
 			TransactionContext txnCtx,
+			TransactionRecordService transactionRecordService,
 			GlobalDynamicProperties dynamicProperties
 	) {
-		this.ledger = ledger;
+		this.accountStore = accountStore;
 		this.validator = validator;
 		this.txnCtx = txnCtx;
+		this.transactionRecordService = transactionRecordService;
 		this.dynamicProperties = dynamicProperties;
 	}
 
 	@Override
 	public void doStateTransition() {
-		try {
-			final var op = txnCtx.accessor().getTxn().getCryptoUpdateAccount();
-			final var target = op.getAccountIDToUpdate();
-			final var customizer = asCustomizer(op);
-
-			if (op.hasExpirationTime() && !validator.isValidExpiry(op.getExpirationTime())) {
-				txnCtx.setStatus(INVALID_EXPIRATION_TIME);
-				return;
-			}
-			final var validity = sanityCheck(target, customizer);
-			if (validity != OK) {
-				txnCtx.setStatus(validity);
-				return;
-			}
+		/* --- Translate from gRPC types --- */
+		final var txnBody = txnCtx.accessor().getTxn();
+		final var op = txnBody.getCryptoUpdateAccount();
 
-			ledger.customize(target, customizer);
-			txnCtx.setStatus(SUCCESS);
-		} catch (MissingAccountException mae) {
-			txnCtx.setStatus(INVALID_ACCOUNT_ID);
-		} catch (DeletedAccountException aide) {
-			txnCtx.setStatus(ACCOUNT_DELETED);
-		} catch (Exception e) {
-			log.warn("Unhandled error while processing :: {}!", txnCtx.accessor().getSignedTxnWrapper(), e);
-			txnCtx.setStatus(FAIL_INVALID);
-		}
-	}
-
-	private ResponseCodeEnum sanityCheck(AccountID target, HederaAccountCustomizer customizer) {
-		if (!ledger.exists(target) || ledger.isSmartContract(target)) {
-			return INVALID_ACCOUNT_ID;
-		}
+		/* --- Load the model objects --- */
+		final var accountId = Id.fromGrpcAccount(op.getAccountIDToUpdate());
+		final var accountToBeUpdated = accountStore.loadAccount(accountId);
 
-		final var changes = customizer.getChanges();
-		final var keyChanges = customizer.getChanges().keySet();
+		validateFalse(accountToBeUpdated.isSmartContract(), ACCOUNT_EXPIRED_AND_PENDING_REMOVAL);
```

Comment:
I do not think that this is a valid check. Why would we reject if the account is smart contract? I cannot see this check in the legacy implementation

Reply:
indeed, was obsolete validation